sycl/doc/GetStartedGuide.md

```diff
@@ -280,7 +280,50 @@ command:
 
 ### Test DPC++ toolchain
 
-#### Run regression tests
+There are 3 types of tests which are used for DPC++ toolchain validation:
+* DPC++ in-source LIT tests including [check-llvm](../../llvm/test),
+[check-clang](../../clang/test), [check-llvm-spirv](../../llvm-spirv/test) and
+[check-sycl](../../sycl/test) targets stored in this repository. These tests
+should be not have hardware or external software dependencies (e.g OpenCL,
+Level Zero, CUDA runtimes). All test which do not follow this approach should
+be moved to DPC++ end-to-end tests.
+
+* DPC++ end-to-end (E2E) tests which are extension to
+[LLVM\* test suite](https://github.com/intel/llvm-test-suite/tree/intel/SYCL).
+These tests cover full DPC++ stack including all dependencies (e.g. OpenCL,
+Level Zero, CUDA runtimes).
+
+* SYCL-CTS are official [Khronos\* SYCL\* conformance tests](https://github.com/KhronosGroup/SYCL-CTS).
+They verify SYCL 1.2.1 specification compatibility. All implementation details
+or extensions are out of scope for the tests.
+
+#### Contribution to DPC++ tests
+Every product change should be accompanied with corresponding test modification
+(adding new test(s), extending or modifying existing test(s)).
+
+The test location is selected depending on test nature:
+
+ - A test which depends only on tools and libraries produced during build of
+the repository can be put to in-source LIT. Also the tests for a feature under
+active development requiring atomic change for tests and product can be put to
+[sycl/test/on-device](../../sycl/test/on-device) temporarily. It is developer
+responsibility to move the tests to DPC++ E2E test suite once feature is
+stabilized.
```

Comment:
Does it make sense to rename "on-device" directory to "E2E-unstable" or something like this to reflect the purpose of these tests better?
I also suggest adding a README file to the root of this directory with similar information clarifying the purpose of these tests, expectations we put on developers, etc.

Reply:
I have added README file but would prefer to rename directory as separate PR when there will be agreement on the exact name.